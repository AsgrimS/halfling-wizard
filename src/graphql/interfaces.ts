import {
  Class,
  Subclass,
  Proficiency,
  ClassProficiency_Choices,
  AbilityScoreSkills,
  ClassSpellcasting,
  ClassStarting_Equipment,
} from "./schema";

export interface HeroClassesData {
  classes: Class[];
}

export interface HeroClassData {
  class: Class;
}

export interface SubclassesData {
  subclasses: Subclass[];
}

export interface ProficienciesData {
  proficiencies: Proficiency[];
}

export interface ProficienciesChoicesData {
  proficienciesChoices: ClassProficiency_Choices[];
}
export interface SavingThrowsData {
  saving_throws: AbilityScoreSkills[];
}

export interface ClassSpellcastingData {
  spellcasting: ClassSpellcasting;
}

export interface StartingEquipmentData {
  startingEquipments: ClassStarting_Equipment[];
}
