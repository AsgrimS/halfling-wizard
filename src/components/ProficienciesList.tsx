import React from "react";
import { v4 as uuidv4 } from "uuid";
import { Proficiency, ClassProficiency_Choices } from "../graphql/schema";

interface ProficienciesListProps {
  proficiencies: Proficiency[];
  proficienciesChoices: ClassProficiency_Choices[];
}

const ProficienciesList: React.FC<ProficienciesListProps> = ({
  proficiencies,
  proficienciesChoices,
}) => {
  return (
    <div>
      <span>Proficiencies:</span>
      <ul>
        {proficiencies.map((proficiency) => (
          <li key={proficiency.index}>{proficiency.name}</li>
        ))}
      </ul>
      <span>Choices:</span>
      <br />
      {proficienciesChoices.map(({ from, choose }) => (
        <React.Fragment key={uuidv4()}>
          <span>Choose {choose}</span>
          <ul>
            {from!.map(({ name, index }) => (
              <li key={index}>{name}</li>
            ))}
          </ul>
        </React.Fragment>
      ))}
    </div>
  );
};

export default ProficienciesList;
