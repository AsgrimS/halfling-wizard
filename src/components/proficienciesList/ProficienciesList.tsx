import React from "react";
import { ProficienciesData } from "../../graphql/interfaces";

const ProficienciesList: React.FC<ProficienciesData> = ({ proficiencies }) => {
  return (
    <div data-testid="proficiencies-list">
      <span>Proficiencies:</span>
      <ul>
        {proficiencies.map((proficiency) => (
          <li key={proficiency.index}>{proficiency.name}</li>
        ))}
      </ul>
    </div>
  );
};

export default ProficienciesList;
