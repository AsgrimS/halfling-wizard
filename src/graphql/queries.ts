import { gql } from "@apollo/client";

export const HERO_CLASS = gql`
  query GetHeroesClasses($classIndex: String!) {
    class(filter: { name: $classIndex }) {
      hit_die
      index
      name
      proficiencies {
        index
        name
      }
      saving_throws {
        index
        name
      }
      proficiency_choices {
        choose
        from {
          name
          index
        }
      }
      spellcasting {
        level
        spellcasting_ability {
          index
          name
        }
      }
      starting_equipment {
        equipment {
          name
          index
        }
        quantity
      }
      starting_equipment_options {
        choose
        from {
          equipment {
            name
            index
          }
          quantity
        }
      }
      subclasses {
        index
        name
      }
    }
  }
`;

export const HEROES_CLASSES = gql`
  query GetHeroesClasses {
    classes {
      name
      index
    }
  }
`;
