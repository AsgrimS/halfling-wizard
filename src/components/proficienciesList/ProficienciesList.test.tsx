import React from "react";
import { render, screen, cleanup } from "@testing-library/react";

import ProficienciesList from "./ProficienciesList";
import { Proficiency } from "../../graphql/schema";

describe("ProficienciesList component", () => {
  afterEach(() => {
    cleanup();
  });

  it("should render a list of proficiencies", () => {
    const proficienciesMockData: Proficiency[] = [
      { name: "Pancaking skills", index: "pancaking skills" },
      { name: "Physics", index: "physics" },
    ];

    render(<ProficienciesList proficiencies={proficienciesMockData} />);
    const proficienciesListElement = screen.getByTestId("proficiencies-list");
    const liElements = proficienciesListElement.getElementsByTagName("li");

    expect(proficienciesListElement).toBeInTheDocument();
    expect(liElements.length).toEqual(2);

    for (let i = 0; i < liElements.length; i += 1) {
      expect(liElements[i]).toContainHTML(
        `<li>${proficienciesMockData[i].name}</li>`
      );
    }

    expect(proficienciesListElement).toMatchSnapshot();
  });
});
