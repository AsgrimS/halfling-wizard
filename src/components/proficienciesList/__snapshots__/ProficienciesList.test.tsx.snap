// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProficienciesList component should render a list of proficiencies 1`] = `
<div
  data-testid="proficiencies-list"
>
  <span>
    Proficiencies:
  </span>
  <ul>
    <li>
      Pancaking skills
    </li>
    <li>
      Physics
    </li>
  </ul>
</div>
`;
